e2e/capacity/staking.test.ts

```diff
@@ -293,4 +294,45 @@ describe('Capacity Staking Tests', function () {
       await assert.rejects(failStakeObj.signAndSend(), { name: 'BalanceTooLowtoStake' });
     });
   });
+
+  describe('staking when there are other freezes on the balance', function() {
+    let vesterKeys: KeyringPair;
+    let providerKeys: KeyringPair;
+    let providerId: u64;
+
+    async function assertSpendable(keys: KeyringPair, amount: bigint) {
+      const spendable = await getSpendableBalance(keys);
+      assert.equal(spendable, amount, `Expected spendable ${amount}, got ${spendable}`);
+    }
+
+    async function assertFrozen(keys: KeyringPair, amount: bigint) {
+      const accountInfo = await ExtrinsicHelper.getAccountInfo(keys);
+      assert.equal(accountInfo.data.frozen, amount, `Expected frozen ${amount}, got ${accountInfo.data.frozen}`);
+    }
+
+    before(async function () {
+      vesterKeys = await createAndFundKeypair(fundingSource, 50_000_000n);
+      await assertSpendable(vesterKeys, 49n * BigInt(CENTS)); // less ED
+      await assertFrozen(vesterKeys, 0n);
+      providerKeys = await createAndFundKeypair(fundingSource, 10n * CENTS);
+      providerId = await createMsaAndProvider(fundingSource, providerKeys, "Provider Whale", 10n * DOLLARS);
+    });
+
+    it("succeeds when there is a time-release freeze", async function() {
+      const vestingAmount = 100n * DOLLARS;
+      const schedule: ReleaseSchedule = calculateReleaseSchedule(vestingAmount);
+
+      const vestedTransferTx = ExtrinsicHelper.timeReleaseTransfer(fundingSource, vesterKeys, schedule);
+
+      await assert.doesNotReject(vestedTransferTx.signAndSend());
+      await assertFrozen(vesterKeys, 100n * DOLLARS);
+
+      await assert.doesNotReject(ExtrinsicHelper.stake(vesterKeys, providerId, 80n * DOLLARS).signAndSend());
+
+      const spendable = await getSpendableBalance(vesterKeys);
+      // after txn fees
```

Comment:
Check is done this way to account for variable txn fees due to changing weights.